server/src/main/java/com/objectcomputing/checkins/services/member_skill/skillsreport/SkillsReportServicesImpl.java

```diff
@@ -0,0 +1,169 @@
+package com.objectcomputing.checkins.services.member_skill.skillsreport;
+
+import com.objectcomputing.checkins.services.member_skill.MemberSkillRepository;
+import com.objectcomputing.checkins.services.memberprofile.MemberProfileRepository;
+import com.objectcomputing.checkins.services.member_skill.MemberSkill;
+import com.objectcomputing.checkins.exceptions.BadArgException;
+
+import javax.inject.Singleton;
+import java.util.List;
+import java.util.Set;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.UUID;
+
+@Singleton
+public class SkillsReportServicesImpl implements SkillsReportServices {
+    private final MemberSkillRepository memberSkillRepo;
+    private final MemberProfileRepository memberProfileRepo;
+
+    public SkillsReportServicesImpl(MemberSkillRepository memberSkillRepo,
+                                    MemberProfileRepository memberProfileRepo) {
+        this.memberSkillRepo = memberSkillRepo;
+        this.memberProfileRepo = memberProfileRepo;
+    }
+
+    public SkillsReportResponseDTO report(SkillsReportRequestDTO request) {
```

Comment:
we could use a @notnull here instead of manual check on line 28

Reply:
Does `@NotNull` enforce that the parameter cannot be null? Or is it only for documentation purposes?

Reply:
Yes, it enforces that parameter should not be null.

https://docs.oracle.com/javaee/7/api/javax/validation/constraints/NotNull.html

example in repo - https://github.com/oci-labs/check-ins/blob/develop/server/src/main/java/com/objectcomputing/checkins/services/checkin_notes/CheckinNoteServicesImpl.java

Reply:
Okay, thanks.

Reply:
I added `@NotNull` and removed the manual check, but one of the tests shows that the method is still called when the input parameter is `null`. So I'm moving back to use manual check until we figure that out.